File: 11-mersenne-num/mersenne_gen.exs
# Elixir program to generate mersenne numbers
# Created by M.V.Harish Kumar on 24-12-2023

{n, _} = IO.gets("Number of mersenne numbers to generate: ") |> Integer.parse
m_list = Enum.map(1..n, &(Integer.pow(2, &1)-1))
IO.puts "The first #{n} mersenne numbers are:"
IO.puts "\t#{inspect(m_list)}"

File: 11-mersenne-num/mersenne_gen.rb
# Ruby program to generate mersenne numbers
# Created by M.V.Harish Kumar on 01-02-2025

print "Number of mersenne numbers to generate: "
n = gets.to_i
mersenne = []
(1..n).each { |i| mersenne << (2**i - 1)}
puts "The first #{n} mersenne numbers are:"
puts "\t#{mersenne.to_s[1...-1]}"
